auth/src/main/java/org/aerogear/mobile/security/SecurityService.java

```diff
@@ -16,70 +17,87 @@
  * Checks can be run individually using {@link #check(SecurityCheckType)} , or can be chained
  * together using an {@link SyncSecurityCheckExecutor} by using {@link #getCheckExecutor()}.
  */
-public class SecurityService implements ServiceModule{
+public class SecurityService implements ServiceModule {
     private final static String TYPE = "security";
 
+    /**
+     * {@link MobileCore}
+     */
     private MobileCore core;
 
     /**
-     * @return
+     * Gets the service type.
+     *
+     * @return {@link String}
      */
     @Override
     public String type() {
         return TYPE;
     }
 
     /**
-     * @param core                 MobileCore instance
-     * @param serviceConfiguration the configuration for the singleThreadService
+     * Configures the security service.
+     *
+     * @param core                 {@link MobileCore} instance
+     * @param serviceConfiguration {@link ServiceConfiguration} for the security service
      */
     @Override
-    public void configure(final MobileCore core, final ServiceConfiguration serviceConfiguration) {
+    public void configure(@NonNull final MobileCore core, @Nullable final ServiceConfiguration serviceConfiguration) {
         this.core = core;
     }
 
     /**
-     * @return
+     * Checks if the service requires a service configuration.
+     * This service does not require a service configuration.
+     *
+     * @return <code>false</code>
      */
     @Override
     public boolean requiresConfiguration() { return false; }
 
     /**
-     *
+     * Invoked when security service needs to be destroyed.
      */
     @Override
     public void destroy() {}
 
     /**
-     * Retrieve a {@link SyncSecurityCheckExecutor} to run multiple {@link SecurityCheckType checks} chained.
+     * Retrieve a check executor that can synchronously  run multiple security checks chained.
      *
-     * @return A new executor.
+     * @return {@link SyncSecurityCheckExecutor}
      */
     public SyncSecurityCheckExecutor getCheckExecutor() {
         return SecurityCheckExecutor.Builder
             .newSyncExecutor(core.getContext()).build();
     }
 
     /**
-     * Retrieve a {@link AsyncSecurityCheckExecutor} to asynchronously run multiple {@link SecurityCheckType checks} chained.
+     * Retrieve a check executor that can asynchronously run multiple security checks chained.
```

Comment:
I'd remove the `chained` here. The executor simply runs the security check. 
The checks can be added with a fluent api, but saying that it executes multiple checks chained is confusing.